lib/alexaSmartHomeV2.js

```diff
@@ -1447,6 +1522,24 @@ function AlexaSH2(adapter) {
                 //writeResponse(request.payload.appliance.applianceId, '%', request.payload.percentageState.value);
                 break;
 
+            case 'SetColorTemperatureRequest':
+                adapter.log.debug('ALEXA SetColorTemperature: ' + request.payload.appliance.applianceId + ' ' + request.payload.colorTemperatureInKelvin + '%');
+                for (i = 0; i < ids.length; i++) {
+                    controlColorTemperature(ids[i], request.payload.colorTemperatureInKelvin, ids.length === 1, function (err) {
+                        if (!--count) {
+                            request.header.name = 'SetColorTemperature';
+
+                            callback({
+                                header: request.header,
+                                payload: {}

```

Comment:
The code for `SetColorRequests` responds with the state it achieved. What does that do? And how does the payload object for this have to look like?

Reply:
In V2 all "Set...Request" are returning a "Set...Confirmation". So you should change "SetColorTemperature" in the request.header.name to "SetColorTemperatureConfirmation".

See here: https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/archive/shv2-smart-home-skill-api-reference#setcolortemperatureconfirmation